src/components/SignUp.js

```diff
@@ -0,0 +1,173 @@
+import React, { Component } from 'react';
+import Form from 'react-bootstrap/Form';
+import { faChevronUp } from '@fortawesome/free-solid-svg-icons';
+import { Link, Redirect } from 'react-router-dom';
+import { FormInput } from './FormInput';
+import './SignUp.css';
+
+export default class SignUp extends Component {
+  constructor(props) {
+    super(props);
+
+    // Initial state
+    this.state = {
+      f_name: '', // Holds first name
+      l_name: '', // Holds last name
+      email: '', // Holds email
+      confirmEmail: '', // Holds email confirmation
+      password: '', // Holds password
+      confirmPassword: '', // Holds password confirmation
+      attemptedSubmit: false // Tells us if the user clicked submit
+    };
+
+    // Bindings
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  // Save input changes to current state
+  // Note: name and value are retrieved from HTML below
+  handleChange(e) {
+    const { name, value } = e.target;
+    this.setState({ [name]: value }); // Square brackets are important for syntax
+  }
+
+  // When form is submit, check basic requirements and make appropriate
+  // call to container if met. TODO: possible refactor
+  handleSubmit(e) {
+    e.preventDefault();
+    this.setState({ attemptedSubmit: true });
+    const {
+      f_name,
+      l_name,
+      email,
+      confirmEmail,
+      password,
+      confirmPassword
+    } = this.state;
+    /* If all fields exist, and confirmations match, attempt sign up
+    This feels redundant. If anyone can suggest a simpler way to
+    do this that would be great */
+    if (
+      f_name &&
+      l_name &&
+      email &&
+      confirmEmail &&
+      password &&
+      confirmPassword
```

Comment:
It looks like one of the few alternatives is the following:

```
function isEmpty(obj) {
    for(var prop in obj) {
        if(obj.hasOwnProperty(prop))
            return false;
    }

    return true;
}
```

This function could be called using `if(isEmpty(this.state))`. Ultimately, though, both work.